test/extensions/filters/network/common/fuzz/uber_per_readfilter.cc

```diff
@@ -0,0 +1,97 @@
+        .WillByDefault(testing::WithArgs<3>(Invoke([&](Grpc::RawAsyncRequestCallbacks& callbacks) {
+          Filters::Common::ExtAuthz::GrpcClientImpl* grpc_client_impl =
+              dynamic_cast<Filters::Common::ExtAuthz::GrpcClientImpl*>(&callbacks);
+          const std::string empty_body{};
+          const auto expected_headers =
+              Filters::Common::ExtAuthz::TestCommon::makeHeaderValueOption({});
+          auto check_response = Filters::Common::ExtAuthz::TestCommon::makeCheckResponse(
+              Grpc::Status::WellKnownGrpcStatus::Ok, envoy::type::v3::OK, empty_body,
+              expected_headers);
+          // Give response to the grpc_client by calling onSuccess().
+          grpc_client_impl->onSuccess(std::move(check_response), span_);
+          return async_request_.get();
+        })));
+
+    EXPECT_CALL(*async_client_factory_, create()).WillOnce(Invoke([&] {
+      return std::move(async_client_);
+    }));
+
+    EXPECT_CALL(factory_context_.cluster_manager_.async_client_manager_,
+                factoryForGrpcService(_, _, _))
+        .WillOnce(Invoke([&](const envoy::config::core::v3::GrpcService&, Stats::Scope&, bool) {
+          return std::move(async_client_factory_);
+        }));
+  }
+}
+
+void UberFilterFuzzer::checkInvalidInputForFuzzer(const std::string& filter_name,
+                                                  Protobuf::Message* config_message) {
+  // System calls such as reading files are prohibited in this fuzzer. Some input that crashes the
+  // mock/fake objects are also prohibited.
+  const std::string name = Extensions::NetworkFilters::Common::FilterNameUtil::canonicalFilterName(
+      std::string(filter_name));
+  if (filter_name == NetworkFilterNames::get().DirectResponse) {
+    envoy::extensions::filters::network::direct_response::v3::Config& config =
+        dynamic_cast<envoy::extensions::filters::network::direct_response::v3::Config&>(
+            *config_message);
+    if (config.response().specifier_case() ==
+        envoy::config::core::v3::DataSource::SpecifierCase::kFilename) {
+      throw EnvoyException(
+          fmt::format("direct_response trying to open a file. Config:\n{}", config.DebugString()));
+    }
+  } else if (filter_name == NetworkFilterNames::get().LocalRateLimit) {
+    envoy::extensions::filters::network::local_ratelimit::v3::LocalRateLimit& config =
+        dynamic_cast<envoy::extensions::filters::network::local_ratelimit::v3::LocalRateLimit&>(
+            *config_message);
+    if (config.token_bucket().fill_interval().seconds() > seconds_in_one_day_) {
+      // Too large fill_interval may cause "c++/v1/chrono" overflow when simulated_time_system_ is
+      // converting it to a smaller unit. Constraining fill_interval to no greater than one day is
+      // reasonable.
+      throw EnvoyException(
+          fmt::format("local_ratelimit trying to set a large fill_interval. Config:\n{}",
```

Comment:
another very small nit: generally we are trying to move away from fmt lib outside of ENVOY_LOG_MISC -- absl::StrCat can be a drop in replacement here

Reply:
Thanks! Fixed.